### ISSUE TITLE
constructor call is detected as function call

### ISSUE THREAD

**Issue by bleis-tift** [fsprojects/FSharpLint#143]
![image](https://cloud.githubusercontent.com/assets/169348/11355147/19635758-9296-11e5-9aa9-13ffbf986c22.png)

`String(...)` is a constructor.
So the following code is not possible to compile.

``` fsharp
map (List.toArray >> String)
```

**Comment by duckmatt** [comment]
Thanks, currently the type checker isn't used in the linter within VFPT making issues like these undetectable - however this should be in progress soon. Also worth nothing that this code is valid in F# 4

**Comment by duckmatt** [comment]
I'm going to close this as going the linter will be supporting only F# 4 and beyond
